src/Models/CarbonChemistry/carbon_chemistry.jl

```diff
@@ -203,10 +203,10 @@ Alternativly, `pH` is returned if `return_pH` is `true`.
 end
 
 # solves `alkalinity_residual` for pH
-solve_for_H(pH, args...) = 10.0 ^ - pH
+solve_for_H(pH::FT, args...) where FT = convert(FT, 10.0) ^ - pH
 
-solve_for_H(::Nothing, params, upper_pH_bound, lower_pH_bound) =
-    find_zero(alkalinity_residual, (10.0 ^ - upper_pH_bound, 10.0 ^ - lower_pH_bound), Bisection(); atol = 1e-10, p = params)
```

Comment:
Remove Bisection() on this line

Reply:
also careful about hardcoded Float64!

Reply:
actually this is the line from the old code which isn't type stable.  The new code should now be type stable.